### ISSUE TITLE
Scanner: Latest development version of ScanCode causes ORT exception when no SPDX license identifier could be found.

### ISSUE THREAD

**Issue by apfelkraut** [heremaps/oss-review-toolkit#2222]
STEPS TO REPRODUCE
1. Configure ORT to use latest version of ScanCode (e.g. 3.1.1.post337.16377111f)
2. Analyse and then scan a project that contains e.g. [acorn-6.3.0](https://github.com/acornjs/acorn/releases/tag/6.3.0)

OBSERVED
- ORT will throw and exception and stop scanning: _Exception in thread "main" java.lang.IllegalStateException: name must not be null at com.here.ort.scanner.scanners.ScanCode.getLicenseId(ScanCode.kt:380)_

EXPECTED
- ORT will successfully process the native scan results of acorn-6.3.0 and continue scanning other libraries

ROOT CAUSE
- ScanCode 3.0.2 writes `"spdx_license_key": ""` in case that no SPDX license key could be identified.
- ScanCode 3.1.1.post337.16377111f writes `"spdx_license_key": null` in case that no SPDX license key could be identified.
- Writing `""` instead of `null` appears to be a bug of ScanCode that has been resolved in the latest development version, according to @pombredanne
- In other words ORT needs to be improved to also handle `null` as valid value for `spdx_license_key`

**Comment by apfelkraut** [comment]
As solution my proposal would be to change scanner/src/main/kotlin/scanners/ScanCode.kt line 380 from `if (name.isEmpty()) {` to `if (name == null || name.isEmpty()) {`.

At least my scans have then completed successfully ;-)

**Comment by sschuberth** [comment]
Thanks @apfelkraut for reporting. Would you mind verifying the fix at https://github.com/heremaps/oss-review-toolkit/pull/2223?
